server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -339,8 +313,29 @@ private DeveloperProducts getDevProductMap(Consumer consumer, String sku) {
         for (ConsumerInstalledProduct ip : consumer.getInstalledProducts()) {
             devProductIds.add(ip.getProductId());
         }
-        List<Product> prods = productAdapter.getProductsByIds(consumer.getOwner(), devProductIds);
-        return new DeveloperProducts(sku, prods);
+
+        Set<Product> products = new HashSet<Product>();
+        for (Product product : productAdapter.getProductsByIds(consumer.getOwner(), devProductIds)) {
+            for (ProductContent pc : product.getProductContent()) {
+                pc.getContent().addOwner(consumer.getOwner());
+            }
+
+            product.addOwner(consumer.getOwner());
+
+            if (sku.equals(product.getId()) &&
+                StringUtils.isEmpty(product.getAttributeValue("support_level"))) {
+
+                // if there is no SLA, apply the default
+                product.setAttribute("support_level", this.DEFAULT_DEV_SLA);
+            }
+
+            // TODO: Is there any guarantee we'll have the product locally by this point? If we have
+            // to create it, this blows up. May be a use case for bringing back the createOrUpdate
+            // method.
```

Comment:
The products here may/may not exist. The marketing product (sku) on a dev pool would never come in as part of a refresh pools. Those products are intended to be 'global' and never associated with an actual subscription. There is also potential for a provided product to not exist as well, if it is net new and hasn't been brought in via a refresh yet.

i **think** we need createOrUpdate here.